torch/csrc/utils/python_compat.h

```diff
@@ -32,6 +32,27 @@ PyCode_GetNFreevars(PyCodeObject* code) {
 #endif
 }
 
+// CPython hides these structured since
+// https://github.com/python/cpython/pull/19494
+#if PY_VERSION_HEX > 0x030900A6 && !defined(Py_BUILD_CORE)
+/* GC information is stored BEFORE the object structure. */
+typedef struct {
+  // Pointer to next object in the list.
+  // 0 means the object is not tracked
+  uintptr_t _gc_next;
+
+  // Pointer to previous object in the list.
+  // Lowest two bits are used for flags documented later.
+  uintptr_t _gc_prev;
+} PyGC_Head;
+
+/* Get an object's GC head */
+static inline PyGC_Head* _Py_AS_GC(PyObject* op) {
+  char* gc = ((char*)op) - sizeof(PyGC_Head);
```

Comment:
Consider using `offsetof` rather than `sizeof` in case compiler decides to do  some padding